terraform/environments/data-platform/api.tf

```diff
@@ -15,14 +15,20 @@ resource "aws_api_gateway_deployment" "deployment" {
     #       resources will show a difference after the initial implementation.
     #       It will stabilize to only change when resources change afterwards.
     redeployment = sha1(jsonencode([
-      aws_api_gateway_resource.upload_data.id,
+      aws_api_gateway_resource.upload_data,
```

Comment:
I noticed there's a comment above that references removing the `.id`, which is out of date now that you've removed it - can you update that as well?

Reply:
I also wonder if it would be worth creating a confluence page with a checklist of things to do when adding a new endpoint. This list seems easy to forget about.

Reply:
The comment is copied direct from terraform documentation around triggers. There is a bit of further information there that might be useful if we want to explore other options to trigger deployments of the api

And agree checklist would be useful as there are a few different bits to do and in a particular order. Have added a card to trello roadmap